pkg/engine/combat/damage.go

```diff
@@ -0,0 +1,100 @@
+package combat
+
+import (
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+func (mgr *Manager) baseDamage(h *info.Hit) float64 {
+	var dmgMap info.DamageMap = h.BaseDamage
+	damage := 0.0
+	for k, v := range dmgMap {
+		switch k {
+		case model.DamageFormula_BY_ATK:
+			damage += v * h.Attacker.ATK()
+		case model.DamageFormula_BY_DEF:
+			damage += v * h.Attacker.DEF()
+		case model.DamageFormula_BY_MAX_HP:
+			damage += v * h.Attacker.MaxHP()
+		case model.DamageFormula_BY_BREAK_DAMAGE:
+			damage += v * breakBaseDamage[h.Attacker.Level()]
+		}
+	}
+	return damage
+}
+
+func (mgr *Manager) bonusDamage(h *info.Hit) float64 {
+	dmg := 1.0
+	// If hit doesn't use break damage equation, adds dmg%
+	// Otherwise, adds break effect%
+	if h.AsPureDamage {
+		dmg += h.Attacker.GetProperty(prop.AllDamagePercent)
+		dmg += h.Attacker.GetProperty(prop.DamagePercent(h.DamageType))
+		if h.AttackType == model.AttackType_DOT {
+			dmg += h.Attacker.GetProperty(prop.DOTDamagePercent)
+		}
+	} else {
```

Comment:
change this from an else to only apply if using the `ByBreakDamage` formula (so if using break formula + not pure damage, both DMG% and break effect apply)